examples/statemachine/src/cli/generator.ts

```diff
@@ -8,196 +8,200 @@ import fs from 'fs';
 import { CompositeGeneratorNode, NL, processGeneratorNode } from 'langium';
 import { State, Statemachine } from '../language-server/generated/ast';
 
-export class StatemachineGenerator {
-    private statemachine: Statemachine;
-    private fileName: string;
-    private destination: string;
-    private fileNode: CompositeGeneratorNode = new CompositeGeneratorNode();
-
-    constructor(statemachine: Statemachine, fileName: string, destination: string | undefined) {
-        this.statemachine = statemachine;
-
-        const fileNameSeq = fileName.replace(/\..*$/, '').replaceAll(/[.-]/g, '').split('/');
-        this.fileName = `${fileNameSeq.pop() ?? 'statemachine'}.cpp`;
-        this.destination = destination ?? `./${fileNameSeq.join('/')}/generated`;
-    }
+export function generateCpp(statemachine: Statemachine, fileName: string, destination: string | undefined): string {
+    const fileNameSeq = fileName.replace(/\..*$/, '').replaceAll(/[.-]/g, '').split('/');
+    ctx = {
+        statemachine,
+        fileName: `${fileNameSeq.pop() ?? 'statemachine'}.cpp`,
+        destination: destination ?? `./${fileNameSeq.join('/')}/generated`,
+        fileNode: new CompositeGeneratorNode()
+    };
+    return generate();
+}
 
-    public generate(): string {
-        this.fileNode.append('#include <iostream>', NL);
-        this.fileNode.append('#include <map>', NL);
-        this.fileNode.append('#include <string>', NL, NL);
+interface GeneratorContext {
+    statemachine: Statemachine;
+    fileName: string;
+    destination: string;
+    fileNode: CompositeGeneratorNode
+}
 
-        this.fileNode.append(`class ${this.statemachine.name};`, NL, NL);
+let ctx: GeneratorContext;
```

Comment:
The generator context shouldn't be a global object, but rather passed between the functions. Every function should accept a `GeneratorContext` as its parameter instead of accessing a global object.